lib/dry/types/options.rb

```diff
@@ -6,9 +6,9 @@ module Options
 
       # @see Definition#initialize
       def initialize(*args, meta: EMPTY_HASH, **options)
-        @__args__ = args
-        @options = options
-        @meta = meta
+        @__args__ = args.freeze
```

Comment:
Please don't do this, dry-types doesn't offer an interface for mutation, this style of defensive programming doesn't really help, quite the opposite, it pollutes code and slows things down.

Reply:
@flash-gordon 

So the `@options` and `@meta` hashes are directly exposed, and can be freely mutated by any caller of `#options` and `#meta`; to preserve immutability, those two need to be frozen.

`@ __args__` is not exposed, but it seemed strange to not freeze it when I was freezing the other two. Also, I don't think it would slow performance; if anything, it would allow the interpreter to make better memory allocation decisions knowing that it won't be mutated.

Reply:
I won't argue on this because it's bikeshedding. But I don't think the interpreter would work faster, there are potential memory gains though but again they can be achieved with a generational GC (and cruby already has it).